Proper asymmetric projects detection

mathjazz opened the pull request:
Instead of checking if project contains at least one resource of any asymmetric format, we now check if all project formats are asymmetric.

Use Resource object's is_asymmetric property.

@Osmose, r?

Osmose commented on pontoon/base/models.py:
```diff
@@ -263,10 +263,12 @@ class Resource(models.Model):
 
     ALLOWED_EXTENSIONS = [f[0] for f in FORMAT_CHOICES] + ['pot']
 
+    ASYMMETRIC = ('dtd', 'properties', 'ini', 'inc', 'l20n')
```
`ASYMMETRIC_FORMATS` is nicer. :D

Osmose commented:
You should rebase those two commits into a single commit.

Is this attached to a bug? If not, why the change?

r+wc on the code though, nice work!

mathjazz commented:
Renamed and rebased.

This is not attached to a bug. The change was needed because Firefox for Android Aurora only uses 3 ASYMMETRIC_FORMATS (not all), so dumps weren't handled properly. Instead of removing all locale files first and than recreating them from source files for which translations existed, we just updated existing files. That means outdated files were not staged for deletion.
